src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -420,6 +421,44 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                             methodBuilder.Add(method);
                         }
                     }
+                    else
+                    // argumentsOnly == true
+                    {
+                        foreach(var argumentName in _argumentNames)
+                        {
+                            bool found = false;
+                            int localIndex = 0;
+                            foreach (var local in _localsForBinding)
+                            {
+                                if (local.Name == argumentName)
+                                {
+                                    AppendLocalAndMethod(localBuilder, methodBuilder, local, container, localIndex, GetLocalResultFlags(local));
+                                    found = true;
+                                    break;
+                                }
+                                localIndex++;
+                            }
+
+                            if (!found)
+                            {
+                                // Method parameters (except those that have been hoisted).
+                                int parameterIndex = m.IsStatic ? 0 : 1;
+                                foreach (var parameter in m.Parameters)
+                                {
+                                    var parameterName = parameter.Name;
+                                    if (parameterName == argumentName &&
+                                        GeneratedNames.GetKind(parameterName) == GeneratedNameKind.None &&
+                                        !IsDisplayClassParameter(parameter))
+                                    {
+                                        AppendParameterAndMethod(localBuilder, methodBuilder, parameter, container, parameterIndex);
+                                        break;
+                                    }
+
+                                    parameterIndex++;
+                                }
+                            }
+                        }
+                    }
```

Comment:
Can we merge the two code paths (`argumentsOnly` and `!argumentsOnly`) or is there a reason to have different ordering of arguments in the two cases?